catalog/core/catalog-core-camelcomponent/src/main/java/ddf/camel/component/catalog/content/ContentProducerDataAccessObject.java

```diff
@@ -220,24 +225,17 @@ public void processHeaders(Map<String, Object> headers, StorageRequest storageRe
 
     private void waitForAvailableSource(CatalogFramework catalogFramework)
             throws SourceUnavailableException {
-        long startTime = System.currentTimeMillis();
-        boolean ready;
-        do {
-            if (timeoutHasBeenExceeded(startTime, 3 * 60 * 1000)) {
-                throw new SourceUnavailableException(
-                        "failed to find an available source within 3 minutes");
-            }
-
-            ready = catalogFramework.getSourceInfo(new SourceInfoRequestLocal(false))
-                    .getSourceInfo()
-                    .stream()
-                    .findFirst()
-                    .get()
-                    .isAvailable();
-        } while (!ready);
-    }
-
-    private boolean timeoutHasBeenExceeded(long startTimeInMs, long timeoutInMs) {
-        return System.currentTimeMillis() - startTimeInMs > timeoutInMs;
+        try {
+            WaitCondition.expect("Source should be available")
+                    .checkEvery(3, TimeUnit.MINUTES)
```

Comment:
You want this long a delay between checks? There's both a period and max time (`checkEvery` and `within`). I think you want this to be `.within(3, TimeUnit.MINUTES)` and then a polling interval something like `.checkEvery(15, TimeUnit.SECONDS)` (or similar).